cub/test/catch2_test_device_partition_flagged.cu

```diff
@@ -0,0 +1,351 @@
+
+CUB_TEST("DevicePartition::Flagged can run with empty input", "[device][partition_flagged]", types)
+{
+  using type = typename c2h::get<0, TestType>;
+
+  const int num_items = 0;
+  thrust::device_vector<type> in(num_items);
+  thrust::device_vector<type> out(num_items);
+  thrust::device_vector<char> flags(num_items);
+
+  // Needs to be device accessible
+  thrust::device_vector<int> num_selected_out(1, 0);
+  int    *d_num_selected_out = thrust::raw_pointer_cast(num_selected_out.data());
+
+  partition_flagged(in.begin(),
+                    flags.begin(),
+                    out.begin(),
+                    d_num_selected_out,
+                    num_items);
+
+  REQUIRE(num_selected_out[0] == 0);
+}
+
+CUB_TEST("DevicePartition::Flagged handles all matched", "[device][partition_flagged]", types)
+{
+  using type = typename c2h::get<0, TestType>;
+
+  const int num_items = GENERATE_COPY(take(2, random(1, 1000000)));
+  thrust::device_vector<type> in(num_items);
+  thrust::device_vector<type> out(num_items);
+  c2h::gen(CUB_SEED(2), in);
+
+  thrust::device_vector<char> flags(num_items);
+  thrust::fill(flags.begin(), flags.end(), 1);
+
+  // Needs to be device accessible
+  thrust::device_vector<int> num_selected_out(1, 0);
+  int    *d_num_selected_out = thrust::raw_pointer_cast(num_selected_out.data());
+
+  partition_flagged(in.begin(),
+                    flags.begin(),
+                    out.begin(),
+                    d_num_selected_out,
+                    num_items);
+
+  REQUIRE(num_selected_out[0] == num_items);
+  REQUIRE(out == in);
+}
+
+CUB_TEST("DevicePartition::Flagged handles no matched", "[device][partition_flagged]", types)
+{
+  using type = typename c2h::get<0, TestType>;
+
+  const int num_items = GENERATE_COPY(take(2, random(1, 1000000)));
+  thrust::device_vector<type> in(num_items);
+  thrust::device_vector<type> out(num_items);
+  c2h::gen(CUB_SEED(2), in);
+
+  thrust::device_vector<char> flags(num_items);
+  thrust::fill(flags.begin(), flags.end(), 0);
+
+  // Needs to be device accessible
+  thrust::device_vector<int> num_selected_out(1, 0);
+  int    *d_num_selected_out = thrust::raw_pointer_cast(num_selected_out.data());
+
+  partition_flagged(in.begin(),
+                    flags.begin(),
+                    out.begin(),
+                    d_num_selected_out,
+                    num_items);
+
+  // The false partition is in reverse order
+  thrust::reverse(out.begin(), out.end());
+
+  REQUIRE(num_selected_out[0] == 0);
+  REQUIRE(out == in);
+}
+
+CUB_TEST("DevicePartition::Flagged does not change input", "[device][partition_flagged]", types)
+{
+  using type = typename c2h::get<0, TestType>;
+
+  const int num_items = GENERATE_COPY(take(2, random(1, 1000000)));
+  thrust::device_vector<type> in(num_items);
+  thrust::device_vector<type> out(num_items);
+  c2h::gen(CUB_SEED(2), in);
+
+  const int num_selected                    = get_num_selected(num_items);
```

Comment:
Any reason not to use `GENERATE_COPY(take(1, random(1, num_items - 1)))`?

Reply:
This has become obsolete, because we determine the number of selected items from the randomly generated flags